### PR TITLE
chore: clean up folly lib def for bazel builds

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: Marie Bremner <marwhal@fb.com>

## Summary
Got inspired by  https://github.com/magma/magma/pull/9224 and tried cleaning up the folly def

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: third_party/system_libraries.BUILD [hunk 1]
```diff
@@ -21,14 +21,16 @@ config_setting(
...
-        ":use_folly_so": ["usr/local/lib/libfolly.so"],
+        ":use_folly_so": [],
         "//conditions:default": [
             "usr/local/lib/libfolly.a",
             "usr/local/lib/libfmt.a",
         ],
     }),
     linkopts = select({
         ":use_folly_so": [
+            "-L/usr/local/lib",
```
Comment (electronjoe) [inline third_party/system_libraries.BUILD:32]: Can you explain why this was necessary? Or was it trial and error? I would have thought my `ldconfig` would fix this.. maybe.

**Review by electronjoe — APPROVED** [review]

File: third_party/system_libraries.BUILD (hunk 1 above)
Comment (themarwhal) [reply third_party/system_libraries.BUILD:32]: Trial and error. I ran ldconfig but that didn't seem to help

Comment (electronjoe) [reply third_party/system_libraries.BUILD:32]: Can you show me the error you get when you lack -L/usr/locallib?

Comment (themarwhal) [reply third_party/system_libraries.BUILD:32]: so just checked that the devcontainer works without "-L/usr..."
in the Magma VM, let me try agian..

Comment (themarwhal) [reply third_party/system_libraries.BUILD:32]: scratched that this is the error even on the devcontainer: `/usr/bin/ld.gold: error: cannot find -lfolly`

**Pull request closed without merging**